src/AutoRest.CSharp/Mgmt/Decorator/ExtensionDetection.cs

```diff
@@ -17,7 +18,14 @@ public static bool IsExtensionResource(this OperationGroup operationGroup, MgmtC
             if (_valueCache.TryGetValue(operationGroup, out result))
                 return result;
 
-            result = IsExtension(operationGroup, config);
+            if (config.OperationGroupIsExtension.Contains(operationGroup.Key))
```

Comment:
This may be clearer as:

```csharp
result = config.OperationGroupIsExtension.Contains(operationGroup.Key)) ?? IsExtension(operationGroup, config);
```

Reply:
I'm getting this error:

![image](https://user-images.githubusercontent.com/45216704/119557853-91bec300-bd55-11eb-8e97-0be9edf05d09.png)

Reply:
Ah, my mistake. I forgot how ?? worked with non-nullable. 

I meant:

```
result = config.OperationGroupIsExtension.Contains(operationGroup.Key)) || IsExtension(operationGroup, config);
```

But that might be a tad less obvious. Your call.